WooCommerce/WooCommerceTests/ViewRelated/Orders/OrdersViewModelTests.swift

```diff
@@ -28,9 +29,14 @@ final class OrdersViewModelTests: XCTestCase {
     override func setUp() {
         super.setUp()
         storageManager = MockupStorageManager()
+        stores = MockupStoresManager(sessionManager: .makeForTesting(authenticated: true))
+        stores.sessionManager.setStoreId(siteID)
     }
 
     override func tearDown() {
+        // If not resetting store ID back to `nil`, it could cause other test failures since `setStoreId` changes UserDefaults.
+        stores.sessionManager.setStoreId(nil)
```

Comment:
Yeah. Even though we use a different instance, they're still using the same `UserDefaults`. 🙈 So it's still technically a singleton! I wonder how we can improve this. 🤔

Reply:
yea it's a tricky singleton in unit tests that could affect the rest of the tests 😞 I think a straightforward improvement is to call `setStoreId(nil)` or reset `SessionManager.SessionSettings.defaults: UserDefaults` in `SessionManager.makeForTesting` (and we should always use this instead of `SessionManager.testingInstance` https://github.com/woocommerce/woocommerce-ios/issues/2885),  though I haven't tested. Another way is to move away from using the site ID from `SessionManager` and DI the site ID as much as possible.

Reply:
We could probably add a `SessionManagerProtocol` and provide a `MockSessionManager` in unit tests. The mock will not use `UserDefaults`, just in-memory storage. 

And then we'll inject the `SessionManagerProtocol` directly instead of the `StoresManager`. I think it's better if the session and the store are separate entities. 

And yeah, injecting `siteID` is probably better since most views don't need to _react_ to session changes. Injecting `siteID` is how it's usually done in WPAndroid: https://github.com/wordpress-mobile/WordPress-Android/search?q=siteid

Reply:
ah yea, refactoring `SessionManager` to a protocol with a mock implementation using in-memory storage would be the ultimate fix. I created an issue for this: https://github.com/woocommerce/woocommerce-ios/issues/2917